woocommerce/woocommerce
# Track core profiler plugin installation in PHP

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #39664 

This PR adds `coreprofiler_store_extensions_installed_and_activated` track in PHP when plugins are installed via an action scheduler job.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site -- use `Want more options?` and select `Include WooCommerce Beta Tester`
2. Install and activate WooCommerce 8.2.0-a.1 
3. Start the new core profiler. 
4. Open a new tab and navigate to `Tools -> Plugin File Editor`
5. Choose `WooCommerce` from the dropdown and click the select button.
6. Navigate to `src/Admin/PluginsHelper.php` file
7. Add `sleep(31);` at the line 196 
8. Click on `Update file` button
9. Switch back to the core profiler tab.
10. Proceed to the plugins step.
11. Click on `Continue` button on the plugins step.
12. Within 30 seconds, switch back to the plugin file editor and remove `sleep(31)` and click on `Update file` button again.
![Screen Shot 2023-09-14 at 3 45 42 PM](https://github.com/woocommerce/woocommerce/assets/4723145/41ad5557-f22f-452b-9369-01a9d51f2200)


13. Wait for the core profiler to complete and go to  `WooCommerce -> Status -> Scheduled Actions` and make sure `woocommerce_plugins_install_and_activate_async_callback` hook exists. 
14. Wait for `woocommerce_plugins_install_and_activate_async_callback` to complete.
15. Go to `WooCommerce -> Status -> Logs` and click `tracks-2023-MM-DD` log (choose the latest) and click `View`
16. Search for `wcadmin_coreprofiler_store_extensions_installed_and_activated` and make sure it exists. 





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
